src/dotnet-install.ps1

```diff
@@ -154,7 +177,7 @@ function Invoke-With-Retry([ScriptBlock]$ScriptBlock, [int]$MaxAttempts = 3, [in
 
     while ($true) {
         try {
-            return $ScriptBlock.Invoke()
+            return & $ScriptBlock
```

Comment:
Is this properly showing the exceptions on the correct line?

Reply:
Yes. Unlike the other one. 

```powershell
$sb = { 
    throw "hello!"
}

try {
    & $sb
} catch {
    $_ | Out-String
}

try { 
    $sb.Invoke()
} catch {
    $_ | Out-String
}
```
```shell
Exception: untitled:Untitled-1:2:5
Line |
   2 |      throw "hello!"
     |      ~~~~~~~~~~~~~~
     | hello!



MethodInvocationException: untitled:Untitled-1:12:5
Line |
  12 |      $sb.Invoke()
     |      ~~~~~~~~~~~~
     | Exception calling "Invoke" with "0" argument(s): "hello!"
```